papers/scipy/scipy.rst

```diff
@@ -0,0 +1,1706 @@
+The first three items in a row are the coefficients of the numerator of the
+biquad's transfer function, and the second three items are the coefficients
+of the denominator.
+
+The SOS format for an IIR filter is more numerically stable than the
+transfer function format, so it should be preferred when using filters
+with orders beyond, say, 7 or 8.
+
+A disadvantage of the SOS format is that the function ``sosfilt`` (at
+least at the time of this writing) applies an SOS filter by making
+multiple passes over the data, once for each second order section.
+Some tests with, for example, an order 8 filter show that
+``sosfilt(sos, x)`` can require more than twice the time of
+``lfilter(b, a, x)``.
+
+Here we create a Butterworth filter using the SOS representation::
+
+    >>> sos = butter(6, 0.125, output="sos")
+    >>> sos
+    array([[  2.883e-05,   5.765e-05,   2.883e-05,
+              1.000e+00,  -1.349e+00,   4.602e-01],
+           [  1.000e+00,   2.000e+00,   1.000e+00,
+              1.000e+00,  -1.454e+00,   5.741e-01],
+           [  1.000e+00,   2.000e+00,   1.000e+00,
+              1.000e+00,  -1.681e+00,   8.198e-01]])
+
+The array ``sos`` has shape (3, 6).  Each row represents a biquad;
+for example, the transfer function of the biquad stored in the last row is
+
+.. math::
+
+    H(z) = \frac{1 + 2z^{-1} + z^{-2}}{1 - 1.681 z^{-1} + 0.8198 z^{-2}}
+
+**Converting between representations.**
+The ``signal`` module provides a collection of functions for
+converting one representation to another::
+
+    sos2tf, sos2zpk, ss2tf, ss2zpk,
+    tf2sos, tf2zz, tf2zpk, zpk2sos, zpk2ss, zpk2tf 
+
+For example, ``zpk2sos``
+converts from the ZPK representation to the SOS representation.
+In the following, ``z``, ``p`` and ``k`` have the values defined earlier::
+
+    >>> from scipy.signal import zpk2sos
+    >>> zpk2sos(z, p, k) 
+    array([[  2.883e-05,   5.765e-05,   2.883e-05,
+              1.000e+00,  -1.349e+00,   4.602e-01],
+           [  1.000e+00,   2.000e+00,   1.000e+00,
+              1.000e+00,  -1.454e+00,   5.741e-01],
+           [  1.000e+00,   2.000e+00,   1.000e+00,
+              1.000e+00,  -1.681e+00,   8.198e-01]])
+
+
+**Limitations of the transfer function representation.**
+Earlier we said that the transfer function representation of
+moderate to large order IIR filters can result in numerical problems.
+Here we show an example.
+
+We consider the design of a Butterworth bandpass filter
+with order 10 with normalized pass band cutoff frequencies of 0.04
+and 0.16.::
+
+    >>> b, a = butter(10, [0.04, 0.16], btype="bandpass")
+
+We can compute the step response of this filter by applying it to
+an array of ones::
+
+    >>> x = np.ones(125)
+    >>> y = lfilter(b, a, x)
+    >>> plt.plot(y)
+
+The plot is shown in Figure :ref:`fig-unstable-butterworth`.
+(We haven't shown all the additional ``matplotlib`` function calls that
+we used to annotate the plot.)
```

Comment:
Maybe this comment can go together with the one about ``setprintoptions`` for IPython near the start of the chapter, and refer to the companion website or Github repo where all full examples can be found.

Reply:
Yes, I was planning on something like that.  This was written before the later sections.  You can see that later, I switch to just giving the essential code instead of showing commands in an interactive session.  On my to-do list is to update this section to use a similar style.  That change might make it unnecessary to fiddle with `set_printoptions`.